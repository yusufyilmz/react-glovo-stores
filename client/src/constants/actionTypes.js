export const FETCH_CATEGORIES_SUCCESS = 'FETCH_CATEGORIES_SUCCESS';
export const LOADING_CATEGORIES_SUCCESS = 'LOADING_CATEGORIES_SUCCESS';
export const FETCH_CATEGORIES_ERROR = 'FETCH_CATEGORIES_ERROR';

export const FETCH_STORE_SUCCESS = 'FETCH_STORE_SUCCESS';
export const LOADING_STORE_SUCCESS = 'LOADING_STORE_SUCCESS';
export const FETCH_STORE_ERROR = 'FETCH_STORE_ERROR';

export const FILTER_STORE_SUCCESS = 'FILTER_STORE_SUCCESS';
export const FILTER_STORE_ERROR = 'FILTER_STORE_ERROR';

