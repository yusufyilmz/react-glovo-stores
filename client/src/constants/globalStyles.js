import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  body {
    font-family: Gotham,Arial,Helvetica Neue,Helvetica,sans-serif;
    font-weight: 300;
    font-size: 14px;
    color: #4a4a4a;
    -webkit-font-smoothing: antialiased;
    -webkit-overflow-scrolling: touch;
    -webkit-tap-highlight-color: transparent;
    overflow-x: hidden;
  }
`;

export const primaryColor = '#2abb9b'

export const secondColor = '#D3D3D3'
