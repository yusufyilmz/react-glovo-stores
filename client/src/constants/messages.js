export const NO_STORE = 'No store found matching this tag';
export const ERROR_OCCURED = 'There seems to be a problem here';
