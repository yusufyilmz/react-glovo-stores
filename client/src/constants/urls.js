export const FILTER_ICON_URL = 'https://img.icons8.com/ios/26/000000/search-filled.png'