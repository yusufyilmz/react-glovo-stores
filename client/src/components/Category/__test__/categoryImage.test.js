/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import CategoryCardImageWrapper from '../categoryImage';
import {FILTER_ICON_URL} from '../../../constants/urls';

describe('CATEGORY CARD IMAGE COMPONENT', () => {
  let wrapper; let
    props;

  beforeEach(() => {
    props = {
      item: {
        closed: true,
        sleepIcon: FILTER_ICON_URL,
        label: 'label'
      }
    };

    wrapper = shallow(
      <CategoryCardImageWrapper {...props} />
    );
  });


  test('renders element correctly', () => {
    expect(wrapper.find('CategoryCardImageWrapper').length).toBe(1);
  });

  test('renders props correctly', () => {
    expect(wrapper.find('CategoryCardImageWrapper').prop('src')).toEqual(props.item.sleepIcon);
  });


    test('snapchat correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
});
