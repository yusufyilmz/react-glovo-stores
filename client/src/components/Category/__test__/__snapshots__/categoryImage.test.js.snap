// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CATEGORY CARD IMAGE COMPONENT snapchat correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardImage
    item={
      Object {
        "closed": true,
        "label": "label",
        "sleepIcon": "https://img.icons8.com/ios/26/000000/search-filled.png",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "alt": "Category label image",
      "as": "img",
      "src": "https://img.icons8.com/ios/26/000000/search-filled.png",
      "variant": "top",
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": true,
        "rules": Array [
          "
    height: 170px;
    position: relative;
    object-fit: cover;
",
        ],
      },
      "defaultProps": Object {
        "as": "img",
        "variant": null,
      },
      "displayName": "CategoryCardImageWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "img",
          "variant": null,
        },
        "displayName": "CardImg",
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "alt": "Category label image",
        "as": "img",
        "src": "https://img.icons8.com/ios/26/000000/search-filled.png",
        "variant": "top",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "rules": Array [
            "
    height: 170px;
    position: relative;
    object-fit: cover;
",
          ],
        },
        "defaultProps": Object {
          "as": "img",
          "variant": null,
        },
        "displayName": "CategoryCardImageWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "img",
            "variant": null,
          },
          "displayName": "CardImg",
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
